### DISCUSSION TITLE
API to check if fips module is successfully loaded

### DISCUSSION THREAD

**Discussion by aniljadaun** [openssl/openssl#21836] [Q&A] [answered]
EVP_default_properties_is_fips_enabled return true if fips is set as default property . 
This returns true irrespective of fips module is loaded successfuly or not.

I am looking for API which will returns the status of fips module loading.

Please suggest.

**Comment by mattcaswell** [comment, marked as answer]
You can use `OSSL_PROVIDER_available` for this purpose, i.e. `if (OSSL_PROVIDER_available(libctx, "fips")) ...`

https://www.openssl.org/docs/man3.1/man3/OSSL_PROVIDER_available.html
